RubberduckTests/Symbols/DeclarationFinderTests.cs

```diff
@@ -1303,6 +1304,667 @@ End Sub
             }
         }
 
+        [Test]
+        [Category("Resolver")]
+        [Category("Interfaces")]
+        public void FindInterfaceImplementationMembersMatchesDeclarationTypes()
+        {
+            var intrface =
+@"Option Explicit
+
+Public Property Get Foo(Bar As Long) As Long
+End Property
+
+Public Property Let Foo(Bar As Long, NewValue As Long)
+End Property
+";
+
+            var implementation =
+@"Option Explicit
+
+Implements TestInterface
+
+Public Property Get TestInterface_Foo(Bar As Long) As Long
+End Property
+
+Public Property Let TestInterface_Foo(Bar As Long, RHS As Long)
+End Property
+";
+            var vbe = new MockVbeBuilder()
+                .ProjectBuilder("UnderTest", ProjectProtection.Unprotected)
+                .AddComponent("TestInterface", ComponentType.ClassModule, intrface, new Selection(1, 1))
+                .AddComponent("TestImplementation", ComponentType.ClassModule, implementation, new Selection(1, 1))
+                .AddProjectToVbeBuilder()
+                .Build();
+
+            using (var state = MockParser.CreateAndParse(vbe.Object))
+            {
+                var declarations = state.DeclarationFinder.AllDeclarations.ToList();
+                var declaration = declarations.Single(decl => decl.DeclarationType == DeclarationType.PropertyGet && decl.IdentifierName.Equals("Foo"));
+
+                var expected = declarations.Single(decl => decl.DeclarationType == DeclarationType.PropertyGet && decl.IdentifierName.Equals("TestInterface_Foo"));
+                var actual = state.DeclarationFinder.FindInterfaceImplementationMembers(declaration).ToList();
+                var results = actual.Count;
+
+                Assert.AreEqual(1, results, "Expected {0} Declarations, received {1}", expected, results);
+                Assert.AreEqual(expected, actual.First(), "Expected {0}, resolved to {1}", expected, actual);
```

Comment:
Shouldn't we also assert the `Let`, too?

Reply:
That's implicit in the first `Assert`, isn't it?

Reply:
Unless I misunderstand the test filters out the `Let` and only looks at the `Get` even though the input code has both `Get` and `Let`.

Reply:
Right, but if there is only 1 result it has to be either a `Get` ***or*** a `Let`, and `expected` is a `PropertyGet`.